core-api/src/test/java/com/optimizely/ab/OptimizelyTest.java

```diff
@@ -148,6 +144,44 @@ public OptimizelyTest(String validDatafile, String noAudienceDatafile) throws Co
         this.datafileVersion = Integer.parseInt(validProjectConfig.getVersion());
     }
 
+    @Test
+    public void testClose() throws IOException {
+        EventHandler mockCloseableEventHandler = mock(
+            EventHandler.class,
+            withSettings().extraInterfaces(Closeable.class)
+        );
+        ProjectConfigManager mockCloseableProjectConfigManager = mock(
+            ProjectConfigManager.class,
+            withSettings().extraInterfaces(Closeable.class)
+        );
+
+        Optimizely optimizely = Optimizely.builder()
+            .withEventHandler(mockCloseableEventHandler)
+            .withConfigManager(mockCloseableProjectConfigManager)
+            .build();
+
+        optimizely.close();
+
+        verify((Closeable) mockCloseableEventHandler).close();
+        verify((Closeable) mockCloseableProjectConfigManager).close();
+    }
+
+    @Test(expected = IOException.class)
```

Comment:
We don't want to throw an exception. I think we should be catching all exceptions.